src/module/system/check/check.ts

```diff
@@ -457,26 +460,48 @@ class CheckPF2e {
 
         let rerollFlavor = game.i18n.localize(`PF2E.RerollMenu.MessageKeep.${keep}`);
 
-        if (heroPoint) {
+        if (heroPoint && !usePoint) {
+            fu.logCompatibilityWarning('The heroPoint option is deprecated. Use usePoint: "hero" instead.', {
+                since: "7.3",
+                until: "7.4",
+            });
+            usePoint = "hero";
+        }
+
+        if (usePoint) {
             const rerollingActor = actor.isOfType("familiar") ? actor.master : actor;
 
-            // If the reroll costs a hero point, first check if the actor has one to spare and spend it
+            // If the reroll costs a hero or mythic point, first check if the actor has one to spare and spend it
             if (rerollingActor?.isOfType("character")) {
-                const heroPointCount = rerollingActor.heroPoints.value;
-                if (heroPointCount) {
-                    await rerollingActor.update({
-                        "system.resources.heroPoints.value": Math.clamp(
-                            heroPointCount - 1,
-                            0,
-                            rerollingActor.heroPoints.max,
-                        ),
-                    });
-                    rerollFlavor = game.i18n.format("PF2E.RerollMenu.MessageHeroPoint", { name: rerollingActor.name });
-                } else {
-                    ui.notifications.warn(
-                        game.i18n.format("PF2E.RerollMenu.WarnNoHeroPoint", { name: rerollingActor.name }),
-                    );
-                    return;
+                const resource = (() => {
+                    switch (usePoint) {
```

Comment:
You can likely simplify this logic by using a format string. The plural name of the resource is in the resource data, which is in turn localized from PF2E.Actor.Resource (which regrettably, I should have keyed by either property name or slug)

You would do something like `resource = resource ?? (hero-points ? "hero-points" : null)` and then get the resource data from `actor.getResource(slug)`. Make sure you use the slugs, even if camel case is technically supported.

Reply:
That is indeed much simpler and cleaner! Updated.